src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -53,39 +53,87 @@ object DeployPlugin extends AutoPlugin {
+    // The universal packager doesn't clean the staging directory by default.
+    val cleanStage = taskKey[Unit]("Clean the staging directory.")
 
-    val nodeEnv = settingKey[String]("The value to use for NODE_ENV during deploy builds.")
-
-    val makeDefaultBashScript = settingKey[Boolean](
-      "If true, will create executable bash script per JavaAppPackaging defaults. " +
-        "Defaults to false"
+    val deploy = inputKey[Unit](
+      s"Deploy this project to a remote host specified in conf/deploy.conf.\n$Usage"
     )
 
-    val filterNotCacheKeyGenFileNames = settingKey[Seq[String]]("starts of jars in stage you don't want hashed for cachekey")
+    // The reason this is a Setting instead of just including * is that including * in the rsync
+    // command causes files created on the server side (like log files and .pid files) to be
+    // deleted when the rsync runs, which we don't want to happen.
+    val deployDirs = settingKey[Seq[String]](
+      "Subdirectories from the stage task to copy during deploy. " +
+        "Defaults to bin/, conf/, lib/, and public/."
+    )
 
-    val cleanStage = taskKey[Unit](
-      "Cleans the staging directory. This is not done by default by the universal packager."
+    val filterNotCacheKeyGenFileNames = settingKey[Seq[String]](
+      "Starts of jars in stage you don't want hashed for cachekey."
     )
 
-    val stageAndCacheKey = taskKey[File]("stages and calculates cacheKey of this project")
+    val gitRepoClean = taskKey[Unit]("Assert that this project's git repository is clean.")
 
-    /** The reason this is a Setting instead of just including * is that including * in the rsync
-      * command causes files created on the server side (like log files and .pid files) to be
-      * deleted when the rsync runs, which we don't want to happen.
-      */
-    val deployDirs = settingKey[Seq[String]](
-      "subdirectories from the stage task to copy during deploy. " +
-        "defaults to bin/, conf/, lib/, and public/"
+    val gitRepoPresent = taskKey[Unit]("Assert that there is a git repository in the cwd")
+
+    val makeDefaultBashScript = settingKey[Boolean](
+      "If true, will create executable bash script per JavaAppPackaging defaults. " +
+        "Defaults to false."
     )
 
-    val gitRepoClean = taskKey[Unit]("Succeeds if the git repository is clean")
+    // TODO(danm): Create a Deploy scope to reduce confusion around duplication of this setting
+    // between this plugin and the NodeJsPlugin.
+    val nodeEnv = settingKey[String]("The value to use for NODE_ENV during deploy builds.")
+
+    val stageAndCacheKey = taskKey[File]("Stage the current project, and calculate its cache key.")
 
-    val gitRepoPresent = taskKey[Unit]("Succeeds if a git repository is present in the cwd")
   }
 
   import autoImport._
 
+  /** Default settings and task dependencies for the Keys defined by this plugin. */
+  override def projectSettings: Seq[Def.Setting[_]] = Seq(
```

Comment:
Moved this up top because it was just floating in the midst of the deploy helper functions.